### PR TITLE
Twitch chat protocol upgrades

### PR DESCRIPTION
**Pull request by mrphlip**
* Up IRC library version to include handling for IRCv3 tags
* Send CAP REQ messages on connect
* Get message metadata from IRCv3 tags instead of from "jtv" user
 * NB: still getting notifications from "twitchnotify" user, that doesn't seem to have changed
* Update chatlog to use the new way of getting the emotes
 * NB: still using the old way of generating the emotes for *outgoing* messages

### CODE REVIEW

File: lrrbot/__init__.py [hunk 1]
```diff
@@ -201,18 +201,23 @@ def wrapper(target, message):
...
+		metadata = {
+			'usercolor': tags.get('color'),
+			'emotes': tags.get('emotes'),
+			'display-name': tags.get('display-name') or nick,
+			'specialuser': set(),
+		}
+		if int(tags.get('subscriber', 0)):
+			metadata['specialuser'].add('subscriber')
+		if int(tags.get('turbo', 0)):
+			metadata['specialuser'].add('subscriber')
```
Comment (andreasots) [inline lrrbot/__init__.py]: Shouldn't this be `metadata['specialuser'].add('turbo')`?

File: lrrbot/__init__.py [hunk 2]
```diff
@@ -172,7 +170,9 @@ def wrapper(function):
 	def on_connect(self, conn, event):
 		"""On connecting to the server, join our target channel"""
 		log.info("Connected to server")
-		self.twitchclient(conn, 3)
+		conn.cap("REQ", "twitch.tv/membership") # get join/part messages
```
Comment (andreasots) [inline lrrbot/__init__.py]: The only thing we use `JOIN`/`PART` for is checking for `:lrrbot JOIN #loadingreadyrun` and that is sent even without this capability.

File: lrrbot/__init__.py (hunk 1 above)
Comment (mrphlip) [inline lrrbot/__init__.py]: Good catch... fixed

File: lrrbot/__init__.py (hunk 2 above)
Comment (mrphlip) [inline lrrbot/__init__.py]: Yeah... there's a couple of parts of the bot that don't work without join/part messages, in particular the mode tracking... but we're doing our own tracking of who's +o for other reasons anyway, and the bot's been working fine for some time without join/part messages, so I guess it isn't really needed.

**Comment by andreasots** [comment]
Looks good, merging.

**Pull request merged**